sdks/java/io/cdap/src/main/java/org/apache/beam/sdk/io/cdap/Plugin.java

```diff
@@ -83,46 +86,50 @@ public Plugin withConfig(PluginConfig pluginConfig) {
    * validating connection to the CDAP sink/source and performing initial tuning.
    */
   public void prepareRun() {
-    PluginConfig pluginConfig = getPluginConfig();
-    checkStateNotNull(pluginConfig, "PluginConfig should be not null!");
-    if (cdapPluginObj == null) {
-      try {
-        Constructor<?> constructor =
-            getPluginClass().getDeclaredConstructor(pluginConfig.getClass());
-        constructor.setAccessible(true);
-        cdapPluginObj = (SubmitterLifecycle) constructor.newInstance(pluginConfig);
-      } catch (Exception e) {
-        LOG.error("Can not instantiate CDAP plugin class", e);
-        throw new IllegalStateException("Can not call prepareRun");
-      }
-    }
-    try {
-      cdapPluginObj.prepareRun(getContext());
-      if (getPluginType().equals(PluginConstants.PluginType.SOURCE)) {
-        for (Map.Entry<String, String> entry :
-            getContext().getInputFormatProvider().getInputFormatConfiguration().entrySet()) {
-          getHadoopConfiguration().set(entry.getKey(), entry.getValue());
+    if (!isUnbounded()) {
+      PluginConfig pluginConfig = getPluginConfig();
+      checkStateNotNull(pluginConfig, "PluginConfig should be not null!");
+      if (cdapPluginObj == null) {
+        try {
+          Constructor<?> constructor =
+              getPluginClass().getDeclaredConstructor(pluginConfig.getClass());
+          constructor.setAccessible(true);
+          cdapPluginObj = (SubmitterLifecycle) constructor.newInstance(pluginConfig);
+        } catch (Exception e) {
+          LOG.error("Can not instantiate CDAP plugin class", e);
+          throw new IllegalStateException("Can not call prepareRun");
         }
-      } else {
-        for (Map.Entry<String, String> entry :
-            getContext().getOutputFormatProvider().getOutputFormatConfiguration().entrySet()) {
-          getHadoopConfiguration().set(entry.getKey(), entry.getValue());
+      }
+      try {
+        cdapPluginObj.prepareRun(getContext());
```

Comment:
Basically, we need this try-catch only to deal with this line of code. If you move it to a separate private method, it will increase the readability of the code dramatically.

Reply:
Fixed try-catch block.
Don't think that we need a separate method for this, since our method already called `prepareRun()`.